cmd/kubeadm/app/phases/addons/proxy/manifests.go

```diff
@@ -175,6 +175,20 @@ spec:
         - mountPath: /run/xtables.lock
           name: xtables-lock
           readOnly: false
+	  initContainers:
+      - name: init-kube-proxy
+        image: gcr.io/google_containers/busybox
+        command:
+        - /bin/sh
+        - -c
+        - cat /var/lib/kube-proxy/config.conf | grep "mode: ipvs" &&
+          modprobe -a ip_vs ip_vs_rr ip_vs_wrr ip_vs_sh nf_conntrack_ipv4 ||
+          echo "didn't load ipvs related kernel modules"
```

Comment:
Can we use an if statement and use an explicit success/failure return code?

Reply:
I thought this can be simpler. And I worried if the command `cat /var/lib/kube-proxy/config.conf | grep "mode: ipvs"` return non zero, the init container will exist with error.

Reply:
But if we specified that we want IPVS mode, wouldn't we want that to happen? Otherwise we will log the error but continue on to the regular container and fallback to iptables/userspace.

cc @MrHohn @bowei

Reply:
Yeah, that's the original logic. If can't use ipvs mode, give a warning and slightly fallback to iptables to make sure kube-proxy still can work. 
We can talk about it, to chose a more reasonable logic.
@m1093782566

Reply:
If failed to load kernel modules, IPVS proxier can't work. In order to make kube-proxy work, we let kube-proxy fall back to iptables or userspace mode... That's NOT a silent action, kube-proxy will log the reason.

Reply:
Cite a case, if user specify iptables mode but kernel is not compatible or iptables version too low, kube-proxy will fall back to userspace mode.